src/com/facebook/buck/android/AndroidResource.java

```diff
@@ -157,6 +158,16 @@
 
   private final AtomicReference<String> rDotJavaPackage;
 
+  private final Function<HasAndroidResourceDeps, Path> resourceDirectoryFunction =
+      new Function<HasAndroidResourceDeps, Path>() {
+        @Override
+        public Path apply(HasAndroidResourceDeps input) {
+          return Optional.fromNullable(input.getRes())
+              .transform(getResolver().getPathFunction())
+              .get();
+        }
+      };
+
```

Comment:
With that said, this undoes an optimization @natthu did to require first-order deps for android resources.  The issue you are trying to address is the last part of #328 that I haven't gotten to yet, and it's tricky.

I think the right way to do this is in `AndroidLibraryGraphEnhancer.getBuildableForAndroidResources`, look for any AARs as a dep, and then add the `aar_android_resource` flavor as a dep and all of its deps (because those should be considered as exported deps) so it actually gets packaged.  That should then package up what you need to prevent that compiler error you were seeing.